trio/_core/_run.py

```diff
@@ -509,10 +511,16 @@ def _might_change_registered_deadline(self):
             if old != new:
                 self._registered_deadline = new
                 runner = GLOBAL_RUN_CONTEXT.runner
+                if runner.is_guest:
+                    old_next_deadline = runner.next_deadline()
                 if old != inf:
                     del runner.deadlines[old, id(self)]
                 if new != inf:
                     runner.deadlines[new, id(self)] = self
+                if runner.is_guest:
+                    new_next_deadline = runner.next_deadline()
+                    if old_next_deadline != new_next_deadline:
```

Comment:
Technically I suppose this is only necessary if the new next deadline is sooner than the old one; otherwise it's harmless to wake up early and then go back to sleep. Doesn't much matter though.

Reply:
I *think* you're right? But I couldn't quite convince myself that I understood all the timeout logic (esp. `idle_primed`) well enough to be confident of that, so I guess I'll leave it as-is. Like you say, it's hard to imagine this ever mattering at all.

Though I guess this is a hint that if we can simplify the timeout handling in `unrolled_run`, that would be good :-)